modules/gltf/extensions/audio/gltf_document_extension_audio.cpp

```diff
@@ -0,0 +1,645 @@
+		return state_data_variant;
+	}
+	Array state_data;
+	p_state->set_additional_data(StringName("GLTFAudioData"), state_data);
+	return state_data;
+}
+
+Array _get_or_create_audio_sources_in_state(Ref<GLTFState> p_state) {
+	Variant state_sources_variant = p_state->get_additional_data(StringName("GLTFAudioSources"));
+	if (state_sources_variant.get_type() == Variant::ARRAY) {
+		return state_sources_variant;
+	}
+	Array state_sources;
+	p_state->set_additional_data(StringName("GLTFAudioSources"), state_sources);
+	return state_sources;
+}
+
+Array _get_or_create_audio_emitters_in_state(Ref<GLTFState> p_state) {
+	Variant state_emitters_variant = p_state->get_additional_data(StringName("GLTFAudioEmitters"));
+	if (state_emitters_variant.get_type() == Variant::ARRAY) {
+		return state_emitters_variant;
+	}
+	Array state_emitters;
+	p_state->set_additional_data(StringName("GLTFAudioEmitters"), state_emitters);
+	return state_emitters;
+}
+
+int _get_or_insert_audio_stream_in_state(Ref<GLTFState> p_state, Ref<AudioStream> p_audio_stream) {
+	Array audio_data = _get_or_create_audio_data_in_state(p_state);
+	for (int i = 0; i < audio_data.size(); i++) {
+		Dictionary audio_data_dict = audio_data[i];
+		if (audio_data_dict.get("audio_stream", Variant()) == p_audio_stream) {
+			return i;
+		}
+	}
+	const int new_index = audio_data.size();
+	Dictionary audio_data_dict;
+	// Use snake_case because audio_stream only exists in memory, not in the exported JSON.
+	audio_data_dict["audio_stream"] = p_audio_stream;
+	audio_data.push_back(audio_data_dict);
+	return new_index;
+}
+
+int _get_or_insert_audio_source_in_state(Ref<GLTFState> p_state, Dictionary &p_audio_source) {
+	Array state_sources = _get_or_create_audio_sources_in_state(p_state);
+	for (int i = 0; i < state_sources.size(); i++) {
+		Dictionary state_source = state_sources[i];
+		if (state_source == p_audio_source) {
+			return i;
+		}
+	}
+	const int new_index = state_sources.size();
+	state_sources.push_back(p_audio_source);
+	return new_index;
+}
+
+int _get_or_insert_audio_player_in_state(Ref<GLTFState> p_state, Ref<GLTFAudioPlayer> p_audio_player) {
+	Array state_emitters = _get_or_create_audio_emitters_in_state(p_state);
+	for (int i = 0; i < state_emitters.size(); i++) {
+		Ref<GLTFAudioPlayer> state_emitter = state_emitters[i];
+		if (state_emitter->to_dictionary() == p_audio_player->to_dictionary()) {
+			return i;
+		}
+	}
+	const int new_index = state_emitters.size();
+	state_emitters.push_back(p_audio_player);
+	return new_index;
+}
+
+void _copy_audio_stream_properties_to_audio_source(const Ref<AudioStream> p_audio_stream, Dictionary &p_audio_source) {
+	Ref<AudioStreamWAV> audio_stream_wav = p_audio_stream;
+#ifdef MODULE_MINIMP3_ENABLED
+	Ref<AudioStreamMP3> audio_stream_mp3 = p_audio_stream;
+#endif // MODULE_MINIMP3_ENABLED
+#ifdef MODULE_VORBIS_ENABLED
+	Ref<AudioStreamOggVorbis> audio_stream_ogg = p_audio_stream;
+#endif // MODULE_VORBIS_ENABLED
```

Comment:
The audio formats are conditionally compiled, so you can compile the glTF module without MiniMP3, without Vorbis, or without both if you want to. I tested that this works.